libmproxy/platform/osx.py

```diff
@@ -19,8 +19,16 @@ class Resolver(object):
 
     def original_addr(self, csock):
         peer = csock.getpeername()
-        stxt = subprocess.check_output(self.STATECMD, stderr=subprocess.STDOUT)
-        if "sudo: a password is required" in stxt:
+        insufficient_priv = False
+        try:
+            stxt = subprocess.check_output(self.STATECMD, stderr=subprocess.STDOUT)
+        except subprocess.CalledProcessError, e:
+            if "sudo: a password is required" in e.output:
+                insufficient_priv = True
+            else:
+                raise RuntimeError("Error getting pfctl state: " + repr(e))
```

Comment:
Can we please make an 
```python
except:
    ...
else:
    insufficient_priv = "sudo: a password is required" in stxt
```

block here and then just check for `insufficient_priv`?